dist/src/main/dist/conf/aws-ec2_provision.sh

```diff
@@ -111,8 +111,9 @@ EOL
     args="$args --instance-type $INSTANCE_TYPE"
     args="$args --instance-initiated-shutdown-behavior terminate"
 
-    if [ "$SUBNET_ID" = "default" ]; then
-        args="$args --security-groups $SECURITY_GROUP"
```

Comment:
This is the "odd change" that I was talking about in PR description. It looks like I'm removing the possibility to set the security groups. However, this functionality wasn't actually working anyway and let me explain why.

In the EC2-VPC (which is the default account type for all accounts in EC2 starting from 2013, therefore probably all of the ones that we're targetting), it's mandatory to specify a subnet, in which you want to create your instances. Without it, the `aws` command called later fails with the error:
```
An error occurred (VPCResourceNotSpecified) when calling the RunInstances operation: The specified instance type can only be used in a VPC. A subnet ID or network interface ID is required to carry out the request.
```

Therefore, if you want to provision machines using this provisioner, you have to specify SUBNET_ID with non-empty (or non-default) value, therefore *always* bypassing this branch. Hence, it's safe to remove it as it was never (since 2013) really working.

The probable reason why it was implemented like this was that at the time of the creation, the author used the EC2-Classic, where the subnet was not mandatory. Therefore, this branch could have been entered. But since all of our (in Hazelcast) accounts are now EC2-VPC and it's safe to assume that all the other potential users will have it like this as well, this branch was no longer useful.

To be explicit, this means that there's no support for specifying security groups via this provisioner and should be considered as a missing feature that can be implemented later if needed.